src/formatted-codegen.js

```diff
@@ -0,0 +1,1100 @@
+  "BEFORE_CLASS_ELEMENT",
+  "AFTER_CLASS_ELEMENT",
+  "BEFORE_TERNARY_QUESTION",
+  "AFTER_TERNARY_QUESTION",
+  "BEFORE_TERNARY_COLON",
+  "AFTER_TERNARY_COLON",
+  "AFTER_DO",
+  "BEFORE_DOWHILE_WHILE",
+  "AFTER_DOWHILE_WHILE",
+  "AFTER_FORIN_FOR",
+  "BEFORE_FORIN_IN",
+  "AFTER_FORIN_FOR",
+  "BEFORE_FORIN_BODY",
+  "AFTER_FOROF_FOR",
+  "BEFORE_FOROF_OF",
+  "AFTER_FOROF_FOR",
+  "BEFORE_FOROF_BODY",
+  "AFTER_FOR_FOR",
+  "BEFORE_FOR_INIT",
+  "AFTER_FOR_INIT",
+  "EMPTY_FOR_INIT",
+  "BEFORE_FOR_TEST",
+  "AFTER_FOR_TEST",
+  "EMPTY_FOR_TEST",
+  "BEFORE_FOR_UPDATE",
+  "AFTER_FOR_UPDATE",
+  "EMPTY_FOR_UPDATE",
+  "BEFORE_FOR_BODY",
+  "BEFORE_GENERATOR_STAR",
+  "AFTER_GENERATOR_STAR",
+  "BEFORE_FUNCTION_PARAMS",
+  "BEFORE_FUNCTION_DECLARATION_BODY",
+  "BEFORE_FUNCTION_EXPRESSION_BODY",
+  "BEFORE_ARROW",
+  "AFTER_ARROW",
+  "AFTER_GET",
+  "BEFORE_GET_PARAMS",
+  "BEFORE_GET_BODY",
+  "AFTER_IF",
+  "AFTER_IF_TEST",
+  "BEFORE_ELSE",
+  "AFTER_ELSE",
+  "PARAMETER_AFTER_COMMA",
+  "NAMED_IMPORT_AFTER_COMMA",
+  "IMPORT_AFTER_COMMA",
+  "BEFORE_IMPORT_BINDINGS",
+  "BEFORE_IMPORT_MODULE",
+  "AFTER_IMPORT_BINDINGS",
+  "AFTER_FROM",
+  "BEFORE_IMPORT_NAMESPACE",
+  "BEFORE_IMPORT_STAR",
+  "AFTER_IMPORT_STAR",
+  "AFTER_IMPORT_AS",
+  "AFTER_NAMESPACE_BINDING",
+  "BEFORE_IMPORT_AS",
+  "AFTER_IMPORT_AS",
+  "EXPORTS_AFTER_COMMA",
+  "BEFORE_EXPORT_STAR",
+  "AFTER_EXPORT_STAR",
+  "BEFORE_EXPORT_BINDINGS",
+  "AFTER_EXPORT_BINDINGS",
+  "AFTER_EXPORT",
+  "AFTER_EXPORT_DEFAULT",
+  "BEFORE_EXPORT_AS",
+  "AFTER_EXPORT_AS",
+  "BEFORE_LABEL_COLON",
+  "AFTER_LABEL_COLON",
+  "AFTER_METHOD_GENERATOR_STAR",
+  "AFTER_METHOD_NAME",
+  "BEFORE_METHOD_BODY",
+  "AFTER_NEW",
+  "BEFORE_NEW_ARGS",
+  "EMPTY_NEW_CALL",
+  "RETURN",
+  "AFTER_SET",
+  "BEFORE_SET_PARAMS",
+  "BEFORE_SET_BODY",
+  "BEFORE_CASE_TEST",
+  "AFTER_CASE_TEST",
+  "BEFORE_CASE_BODY",
+  "AFTER_CASE_BODY",
+  "DEFAULT",
+  "AFTER_DEFAULT_BODY",
+  "BEFORE_SWITCH_DISCRIM",
+  "BEFORE_SWITCH_BODY",
+  "THROW",
+  "AFTER_TRY",
+  "BEFORE_CATCH",
+  "BEFORE_FINALLY",
+  "AFTER_FINALLY",
+  "VARIABLE_DECLARATION",
+  "YIELD",
+  "BEFORE_YIELD_STAR",
+  "AFTER_YIELD_STAR",
+  "DECLARATORS_AFTER_COMMA",
+  "BEFORE_INIT_EQUALS",
+  "AFTER_INIT_EQUALS",
+  "AFTER_WHILE",
+  "BEFORE_WHILE_BODY",
+  "AFTER_WITH",
+  "BEFORE_WITH_BODY",
+];
+for (let i = 0; i < separatorNames.length; ++i) {
+  Sep[separatorNames[i]] = {type: separatorNames[i]};
+}
+
+Sep.BEFORE_ASSIGN_OP = function(op) {
+  return {
+    type: "BEFORE_ASSIGN_OP",
+    op: op
+  };
+};
+
+Sep.AFTER_ASSIGN_OP = function(op) {
+  return {
+    type: "AFTER_ASSIGN_OP",
+    op: op
+  };
+};
+
+Sep.BEFORE_BINOP = function(op) {
+  return {
+    type: "BEFORE_BINOP",
+    op: op
+  };
+};
+
+Sep.AFTER_BINOP = function(op) {
+  return {
+    type: "AFTER_BINOP",
+    op: op
+  };
+};
+
+Sep.BEFORE_POSTFIX = function(op) {
+  return {
+    type: "BEFORE_POSTFIX",
+    op: op
+  };
+};
+
+Sep.UNARY = function(op) {
```

Comment:
Unary prefix is grouped with the other unary operators. Unary postfix has its own marker.